lib/Dataflow_algorithms.mli

```diff
@@ -0,0 +1,90 @@
+open Core_kernel
+open Mir
+open Dataflow_types
+
+(**
+   ~~~~~ Still to implement: TODO ~~~~~
+ * Indexed variables are currently handled as monoliths
+ * Dependency functions with `probabilistic_dependence` set to true:
+   * Probably don't correctly traverse the probabilistic graphical model
+   * Shouldn't introduce dependency to non-parameter variables
+ * Variables declared in blocks should go out of scope
+   * This is done already for for-loop index variables
+ * Traverse functions that end in st, since they can change the target
```

Comment:
I think you mean functions that end in `_lp` here. It is a bit of a silly language feature and is questionable whether it should have been in the language at all, but people are using it, so I guess we need to support it.

Reply:
Ah, _lp, thanks.